sync/src/main/java/tech/pegasys/teku/sync/forward/multipeer/BatchSync.java

```diff
@@ -115,7 +117,8 @@ public static BatchSync create(
   @Override
   public SafeFuture<SyncResult> syncToChain(final TargetChain targetChain) {
     final SafeFuture<SyncResult> result = new SafeFuture<>();
-    eventThread.execute(() -> switchSyncTarget(targetChain, result));
+    eventThread.execute(
+        exceptionHandlingRunnable(() -> switchSyncTarget(targetChain, result), result));
```

Comment:
TBH, this construct makes me gut feeling that something needs to be refactored here. I don't consider any specifics though :)

Reply:
Yeah, I was kind of hoping that I could build exception handling into eventThread but couldn't see how.  Generally working with event thread is *way* nicer than trying to chain futures all the time and much safer as you know everything is single threaded, but the exception handling part isn't ideal yet. Will have to ponder it some more and see what we can come up with.